testsuites/syncgateway/functional/tests/test_cbs_collections.py

```diff
@@ -6,27 +6,32 @@
 from libraries.testkit.cluster import Cluster
 from keywords.constants import RBAC_FULL_ADMIN
 from libraries.testkit.admin import Admin
-
+from keywords.exceptions import RestError
 
 # test file shared variables
 bucket = "data-bucket"
+sg_password = "password"
+admin_client = cb_server = sg_username = channels = None
+admin_auth = [RBAC_FULL_ADMIN['user'], RBAC_FULL_ADMIN['pwd']]
 
 
 @pytest.fixture
 def teardown_doc_fixture():
-    def _delete_doc_if_exist(sg_client, url, db, doc_id, auth):
-        if sg_client.does_doc_exist(url, db, doc_id) is True:
-            sg_client.delete_doc(url, db, doc_id, auth=auth)
+    def _delete_doc_if_exist(sg_client, url, db, doc_id, auth, scope, collection):
+        if sg_client.does_doc_exist(url, db, doc_id, scope=scope, collection=collection) is True:
+            sg_client.delete_doc(url, db, doc_id, auth=auth, scope=scope, collection=collection)
     yield _delete_doc_if_exist
 
 
 @pytest.fixture
 def scopes_collections_tests_fixture(params_from_base_test_setup):
     try:  # To be able to teardon in case of a setup error
         # get/set the parameters
-        session_id = None
-        pre_test_db_exists = None
-        pre_test_user_exists = None
+        global admin_client
+        global cb_server
+        global sg_username
+        global channels
```

Comment:
should we really be using global variables? Yield should be enough to get these variables from the fixture.

Reply:
Thanks. Yes, I agree that global variables is not ideal.  I could yield them from the setup, but that would mean a lot of variables coming out, and some might not be used. Balancing between these considerations and making the global (which is a bad practice, I agree), only in the test file, I think I'm slightly preferring them as global. Other views are also valid, of course.